simulariumio/tests/test_agents_from_lists.py

```diff
@@ -0,0 +1,344 @@
+                -442.27202981,
+                202.83568625,
+                -262.13407113,
+                -372.23130078,
+                217.21997368,
+                404.88561338,
+                171.37918011,
+                205.80515525,
+                -65.95336727,
+                0.0,
+                0.0,
+                0.0,
+            ],
+            [
+                245.9111405,
+                372.15936027,
+                -261.94702214,
+                3.50037066,
+                441.92904046,
+                321.75701298,
+                146.23928574,
+                -315.3241668,
+                82.00405173,
+            ],
+            [
+                104.82606074,
+                -413.76671598,
+                366.66127719,
+                136.7228888,
+                -210.69313998,
+                -465.59967482,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+            ],
+        ],
+        [
+            [
+                -148.70447678,
+                225.27562348,
+                -273.51318785,
+                -5.32043612,
+                -55.97783429,
+                413.32948686,
+                165.64239994,
+                322.63703294,
+                -2.2348818,
+                0.0,
+                0.0,
+                0.0,
+            ],
+            [
+                -317.48515644,
+                -237.70246887,
+                238.69661676,
+                94.56942257,
+                346.13786088,
+                -7.93209392,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+            ],
+            [
+                7.77508859,
+                260.16762947,
+                -171.02427873,
+                -20.46326319,
+                179.43194042,
+                485.07810635,
+                0.0,
+            ],
+        ],
+    ]
+
+    list_data = AgentDataLists(
+        times=times,
+        n_agents=n_agents,
+        viz_types=viz_types,
+        unique_ids=unique_ids,
+        types=types,
+        positions=positions,
+        radii=radii,
+        n_subpoints=n_subpoints,
+        subpoints=jagged_subpoints,
+    )
+
+    # make subpoint data not jagged for AgentData
+    subpoints = copy.deepcopy(jagged_subpoints)
+    subpoints[0][0].extend([0.0, 0.0])
+    subpoints[0][2].append(0.0)
+    subpoints[1][1].extend([0.0, 0.0, 0.0])
+    subpoints[2][2].extend([0.0, 0.0, 0.0, 0.0, 0.0])
+
+    agent_data = AgentData(
+        times=np.array(times),
+        n_agents=np.array(n_agents),
+        viz_types=np.array(viz_types),
+        unique_ids=np.array(unique_ids),
+        types=types,
+        positions=np.array(positions),
+        radii=np.array(radii),
+        n_subpoints=np.array([[9, 12, 6], [9, 9, 6], [9, 6, 6]]),
+        subpoints=np.array(subpoints),
+    )
```

Comment:
These tests make me confused about how someone would use this. I assumed from the problem statement that a developer could now send in lists the same way they sent in np arrays and that the code would just handle it. But this looks like the user has to realize they are using lists, which are non standard, and first convert them? Or is this just written out for the tests?

Reply:
Ah interesting. I don't see why we can't do it that way, I just started out this route because it was close to the familiar path of `from_buffer_data()` or `from_dict()`. I like that, I'll go work on that now, having either numpy arrays or lists accepted as input parameters and converting everything to numpy arrays on init

- [ ] get rid of AgentDataLists class and handle list input parameters right in AgentData